### PR TITLE
[cli] install nightly wheels via `ray install-nightly`

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

cc @simon-mo

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/nightly-wheels.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+
```
Comment (simon-mo) [inline python/ray/nightly-wheels.yaml:1]: we probably need to package this file in setup.py

**Review by simon-mo — APPROVED** [review]

File: doc/source/installation.rst [hunk 2]
```diff
@@ -22,7 +22,19 @@ You can install the latest stable version of Ray as follows.
...
 
-Here are links to the latest wheels (which are built for each commit on the
+You can install the latest Ray wheels via the following command:
+
+.. code-block:: bash
+
+  pip install -U ray
+  ray install-nightly
+
+  # Note: the above command may not capture updated library dependencies.
```
Comment (wuisawesome) [inline doc/source/installation.rst:32]: Can you use `.. note` syntax here?

File: python/ray/scripts/scripts.py [hunk 3]
```diff
@@ -1487,6 +1488,57 @@ def globalgc(address):
...
+    raydir = os.path.abspath(os.path.dirname(ray.__file__))
+    all_wheels_path = os.path.join(raydir, "nightly-wheels.yaml")
+
+    wheels = None
+    if os.path.exists(all_wheels_path):
+        with open(all_wheels_path) as f:
+            wheels = yaml.safe_load(f)
+
+    if not wheels:
+        raise click.ClickException(f"Wheels not found in '{all_wheels_path}'!")
```
Comment (wuisawesome) [inline python/ray/scripts/scripts.py:1513]: Can we provide the same URL as below here too?

**Review by wuisawesome — DISMISSED** [review]

File: python/ray/nightly-wheels.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+linux:
```
Comment (robertnishihara) [inline python/ray/nightly-wheels.yaml:1]: Why not just bake this into the script? It seems a bit heavyweight to put it in its own file.

File: python/ray/scripts/scripts.py (hunk 3 above)
Comment (richardliaw) [reply python/ray/scripts/scripts.py:1513]: This isn't a URL; it's a filepath.

Comment (richardliaw) [reply python/ray/scripts/scripts.py:1513]: It's supposed to be on your local machine. The below path is the *specific S3 wheel path*.

File: doc/source/installation.rst (hunk 2 above)
Comment (richardliaw) [reply doc/source/installation.rst]: done

File: python/ray/nightly-wheels.yaml (hunk 4 above)
Comment (richardliaw) [reply python/ray/nightly-wheels.yaml:1]: I didn't think it was appropriate to place these hyperlinks/configuration in our scripts.py entrypoint.

IMO this is like a requirements.txt file, and I don't see a downside to doing it like this. It is also accessible should we want to validate/test/access this later (though I agree this isn't important). 

If you feel strongly, I can change it...

File: python/ray/scripts/scripts.py (hunk 3 above)
Comment (wuisawesome) [reply python/ray/scripts/scripts.py:1513]: Oh what I meant was can we give them the same "Please visit https://docs.ray.io/en/master/installation.html to obtain the latest wheels."

Comment (richardliaw) [reply python/ray/scripts/scripts.py]: ah nice, done

**Review by wuisawesome — APPROVED** [review]

**Pull request merged**